server/src/main/java/org/elasticsearch/action/search/SearchQueryThenFetchAsyncAction.java

```diff
@@ -80,7 +84,11 @@ protected void executePhaseOnShard(final SearchShardIterator shardIt,
                                        final SearchShardTarget shard,
                                        final SearchActionListener<SearchPhaseResult> listener) {
         ShardSearchRequest request = rewriteShardSearchRequest(super.buildShardSearchRequest(shardIt));
-        getSearchTransport().sendExecuteQuery(getConnection(shard.getClusterAlias(), shard.getNodeId()), request, getTask(), listener);
+        Transport.Connection conn = getConnection(shard.getClusterAlias(), shard.getNodeId());
```

Comment:
Can you move this to `AbstractSearchAsyncAction#getConnection` ?  There's nothing that prevents to check the version there.